src/client/datakit_client_9p.ml

```diff
@@ -577,6 +577,8 @@ module Make(P9p : Protocol_9p_client.S) = struct
         (fun hash -> node_of_hash t hash >>*= fn)
 
     let fast_forward t commit =
+      let path = Datakit_path.of_steps_exn (branch_dir t) in
+      FS.make_dirs t.fs ~base:[] path >>*= fun () ->
```

Comment:
`test` doesn't do a create, it just creates the directory:

```
    let create fs name =
      (* Note: DataKit returns success if the branch already exists too,
         so no need to handle errors here. *)
      FS.create_dir fs ~dir:["branch"] name >|*= fun () ->
      { fs; name }
```

The only way I can see it failing is if someone deleted the branch after the create. Would be interesting to see a trace if you can reproduce it.